### PR TITLE
Adds possibility to set user documentation l10n.

### PR DESCRIPTION
**Pull request by pierreozoux**
I still need to figure out how is the documentation shipped in the Nextcloud package, so I can make sure that pt, fr and de are also shipped.

Relates to https://github.com/nextcloud/documentation/issues/216

Thanks for your help!

### CODE REVIEW

**Comment by kesselb** [comment]
Hmm. `nextcloud-17.0.0.zip` is already ~80mb big. Around 25mb are for the english documentation. Could you imagine to offer the documentation for other languages as apps via the app store? We probably need some way how to register the app (e.g. user_manual_de) to the help controller. Probably shipping also english as app might reduce the archive and makes it easier to ship doc updates.

**Comment by pierreozoux** [comment]
@kesselb I just checked, removing images and zipping the user_manual folder, it weights only 35105bits. So adding one language documentation, is just 35kb, adding the ~100language would add 3.5mb. I agree that size does matter, and it is important for various reasons, but I propose that we discuss about it in another issue.

File: apps/settings/lib/Controller/HelpController.php [hunk 1]
```diff
@@ -67,15 +76,35 @@ public function __construct(
 	 * @NoAdminRequired
 	 */
 	public function help(string $mode = 'user'): TemplateResponse {
-		$this->navigationManager->setActiveEntry('help');
+		define('TRANSLATED_LANGUAGES', array(
```
Comment (kesselb) [inline apps/settings/lib/Controller/HelpController.php]: Please make this a class member. e.g: `private const TRANSLATED_LANGUAGES`

Comment (kesselb) [reply apps/settings/lib/Controller/HelpController.php]: I would also prefer a better name like `DOCUMENTATION_LANGUAGES`

**Comment by kesselb** [comment]
Just had a look at `user_manual_de` and it looks outdated to me. We should not ship outdated documentation because this will lead to confusion. Have you thought about the idea of shipping docs for other languages as apps?

**Comment by pierreozoux** [comment]
Agree, I'll remove the DE, and check the brazilian.
We still have to do the french. Anyway, we have to coordinateur between the documentation repot and here.

About the app, Im maybe biased because I don't want to create a repo, create an app, write tests, write a CICD pipeline and maintain an app for this 5 lines PR, but I think Nextcliud has to be consistent. 
I'm not opposed to the idea of splitting into apps, but IMHO, the split has to happen, either at the i18n or the documentation.

Currently Nextcloud ships i18n for the app itself and the documentation. So to be consistent,  I think it has to ship the documentation translated too.
But I wouldn't be mad if Nextcloud decides that i18n or documentation (or both even) is now an optional package.

But again, I think this is another discussion than this PR.

File: apps/settings/lib/Controller/HelpController.php (hunk 1 above)
Comment (pierreozoux) [reply apps/settings/lib/Controller/HelpController.php]: I'll fix when in front of computer, thanks for the advice.

**Review by juliusknorr — CHANGES_REQUESTED** [review]
I'm strongly against doing this until there is a proper way to have translated documentation somehow kept up to date with the English one. Current de and pt_BR didn't receive an update in the last half year.

Besides that, maybe we should find a way to not ship the duplicate documentations (with images) as individual folders but somehow just have the text translations?

**Comment by pierreozoux** [comment]
Well :) the english documentation is not that up to date either ;)

https://github.com/nextcloud/documentation/blob/master/user_manual/whats_new.rst (check the date of the last change)

Im also tackling this translation thing here 

https://github.com/nextcloud/documentation/issues/216#issuecomment-551291335

I'm not asking to merge right away, I'm asking for a bit of support to help me roll the ball around this topic:) thanks for your help :)

The next thing I need is to understand, where is the CI that packages the Nextcloud zip and fetch documentations so I can make sure it fetches the right thing at the right time.

**Comment by ChristophWurst** [comment]
I'd say let's close this for now. This hasn't received any update in quite a while and this is something that needs to be discussed more.

**Comment by pierreozoux** [comment]
@ChristophWurst I'm still waiting https://github.com/nextcloud/documentation/pull/1824 to be merged first.

Can we keep it open? (and help move the other one?)

**Comment by pierreozoux** [comment]
So :)

image

Now we have an automated way to translate documentation \o/

But it is still not shipped inside Nextcloud:

image

(this screenshot is 18.0.14)

I have a simple question, where (I searched) is the line that adds the documentation inside the Nextcloud zip? I just need the answer to that, and then I'll do the necessary PR that is needed to be solved before this PR :)

**Comment by skjnldsv** [comment]
Hey @pierreozoux seems like this PR is quite old and the direction since changed.
As there is no feedback since a while I will close this ticket.

Thanks for the interest in Nextcloud and the effort put into this! 🙇

**Pull request closed without merging**